runtime/development/src/lib.rs

```diff
@@ -1674,6 +1685,94 @@ impl pallet_liquidity_rewards::Config for Runtime {
 	type WeightInfo = ();
 }
 
+const WEIGHT_PER_GAS: u64 = 20_000;
+parameter_types! {
+	pub BlockGasLimit: U256 = U256::from(NORMAL_DISPATCH_RATIO * MAXIMUM_BLOCK_WEIGHT.ref_time() / WEIGHT_PER_GAS);
+	pub PrecompilesValue: CentrifugePrecompiles<Runtime> = CentrifugePrecompiles::<_>::new();
+	pub WeightPerGas: Weight = Weight::from_ref_time(WEIGHT_PER_GAS);
+}
+
+pub struct FindAuthorTruncated<F>(PhantomData<F>);
+impl<F: FindAuthor<u32>> FindAuthor<H160> for FindAuthorTruncated<F> {
+	fn find_author<'a, I>(digests: I) -> Option<H160>
+	where
+		I: 'a + IntoIterator<Item = (ConsensusEngineId, &'a [u8])>,
+	{
+		if let Some(author_index) = F::find_author(digests) {
+			let authority_id = Aura::authorities()[author_index as usize].clone();
+			return Some(H160::from_slice(&authority_id.to_raw_vec()[4..24]));
+		}
+		None
+	}
+}
+
+#[derive(Encode, Decode, Default)]
+struct EthereumAccount(H160);
+
+impl sp_runtime::TypeId for EthereumAccount {
+	const TYPE_ID: [u8; 4] = *b"ETH\0";
+}
+
+pub struct ExtendedAddressMapping;
+impl AddressMapping<AccountId> for ExtendedAddressMapping {
+	fn into_account_id(address: H160) -> AccountId {
+		EthereumAccount(address).into_account_truncating()
+	}
+}
+
+impl pallet_evm::Config for Runtime {
+	type AddressMapping = ExtendedAddressMapping;
+	type BlockGasLimit = BlockGasLimit;
+	type BlockHashMapping = pallet_ethereum::EthereumBlockHashMapping<Self>;
+	type CallOrigin = EnsureAddressTruncated;
+	type ChainId = EVMChainId;
+	type Currency = Balances;
+	type FeeCalculator = BaseFee;
+	type FindAuthor = FindAuthorTruncated<Aura>;
+	type GasWeightMapping = pallet_evm::FixedGasWeightMapping<Self>;
+	type OnChargeTransaction = ();
+	type PrecompilesType = CentrifugePrecompiles<Self>;
+	type PrecompilesValue = PrecompilesValue;
+	type Runner = pallet_evm::runner::stack::Runner<Self>;
+	type RuntimeEvent = RuntimeEvent;
+	type WeightPerGas = WeightPerGas;
+	type WithdrawOrigin = EnsureAddressTruncated;
+}
+
+impl pallet_evm_chain_id::Config for Runtime {}
```

Comment:
How is the chainId value set for an ongoing running chain? Manual raw storage set?

Reply:
It comes from genesis. For adding EVM to a running chain we'll need to implement a migration (we'll also need that to define the "EVM Genesis Block" which will be based on the first block which contains EVM support. Lots of fun :) )

Reply:
Got it 👍